src/relay/pass/well_formed.cc

```diff
@@ -0,0 +1,50 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file well_formed.cc
+ * \brief check that expression is well formed.
+ */
+#include <tvm/relay/pass.h>
+#include <tvm/relay/expr_functor.h>
+#include <unordered_set>
+
+namespace tvm {
+namespace relay {
+
+struct NotWellFormed { };
+
+//! brief make sure each Var is bind at most once.
+struct IsWellForm : ExprVisitor {
+  std::unordered_set<Var> s;
```

Comment:
we can use unordered_set<Var, NodeHash, NodeEqual>

Reply:
should I remove the std hash specialization then?